fendermint/actors/eam/src/state.rs

```diff
@@ -51,16 +53,23 @@ impl State {
         Ok(State { permission_mode })
     }
 
-    pub fn can_deploy(
-        &self,
-        store: &impl Blockstore,
-        deployer: &Address,
-    ) -> Result<bool, ActorError> {
+    pub fn can_deploy(&self, rt: &impl Runtime, deployer: ActorID) -> Result<bool, ActorError> {
         Ok(match &self.permission_mode {
             PermissionMode::Unrestricted => true,
             PermissionMode::AllowList(cid) => {
-                let deployers = DeployerMap::load(store, cid, DEFAULT_HAMT_CONFIG, "verifiers")?;
-                deployers.contains_key(deployer)?
+                let deployers =
+                    DeployerMap::load(rt.store(), cid, DEFAULT_HAMT_CONFIG, "verifiers")?;
+                let mut allowed = false;
+                deployers.for_each(|k, _| {
+                    // Normalize allowed addresses to ID addresses, so we can compare any kind of allowlisted address.
+                    // This includes f1, f2, f3, etc.
+                    // We cannot normalize the allowlist at construction time because the addresses may not be bound to IDs yet (counterfactual usage).
+                    // Unfortunately, API of Hamt::for_each won't let us stop iterating on match, so this is more wasteful than we'd like. We can optimize later.
+                    // Hamt has implemented Iterator recently, but it's not exposed through Map2 (see ENG-800).
+                    allowed |= rt.resolve_address(&k) == Some(deployer);
```

Comment:
NB this does not prevent the execution of `resolve_address` after `allowed` has already been set to `true`.

Reply:
That's a good optimization, let me apply it.